packages/libraries/router/src/registry_logger.rs

```diff
@@ -0,0 +1,98 @@
+use std::{env, fmt};
+
+static LOG_LEVEL_NAMES: [&str; 5] = ["ERROR", "WARN", "INFO", "DEBUG", "TRACE"];
+
+#[repr(usize)]
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Debug)]
+pub enum LogLevel {
+    Error,
+    Warn,
+    Info,
+    Debug,
+    Trace,
+}
+
+impl LogLevel {
+    fn from_usize(u: usize) -> Option<LogLevel> {
+        match u {
+            0 => Some(LogLevel::Error),
+            1 => Some(LogLevel::Warn),
+            2 => Some(LogLevel::Info),
+            3 => Some(LogLevel::Debug),
+            4 => Some(LogLevel::Trace),
+            _ => None,
+        }
+    }
+
+    fn from_str(s: &str) -> LogLevel {
+        LOG_LEVEL_NAMES
+            .iter()
+            .position(|&name| name.eq_ignore_ascii_case(s))
+            .map(|p| LogLevel::from_usize(p).expect("Hive failed to read the log level"))
+            .expect("Hive failed to parse the log level filter")
+    }
```

Comment:
I think here there's a possibility to eliminate an unnecessary panic call, and make it more graceful to failure

```suggestion
    fn from_str(s: &str) -> Option<LogLevel> {
        match s {
            "error" => Some(LogLevel::Error),
            "warn" => Some(LogLevel::Warn),
            "info" => Some(LogLevel::Info),
            "debug" => Some(LogLevel::Debug),
            "trace" => Some(LogLevel::Trace),
            _ => None,
        }
    }
```

Reply:
but we'll also need to modify the `from_str` call down below if you accepted it.